### PR TITLE
[C-1848 ARCH-1146] Refactor search results, Add redux-persist

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Refactors search-results following standard patterns
- Adds redux-persist to handle search history. This greatly simplifies search-history code, since we dont need to fetch and manage saving to async-storage in sagas. Really excited about this option for other parts of our state, theme, prefs etc.
- Fixes case where signing-out did not clear history. So simple now! just call `clearHistory` action and we clear in redux and local-storage.

### Dragons

- Need to ensure PersistGate doesn't degrade loading screen, couldn't see anything in my testing

### CODE REVIEW

File: packages/mobile/src/screens/search-screen/content/EmptySearch.tsx [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
 
 const messages = {
   empty: 'Search for Artists, Tracks, Playlists, & Albums',
   noResults: 'Sorry, we couldn’t find anything that matches'
 }
 
-const styles = StyleSheet.create({
+const useStyles = makeStyles(({ spacing }) => ({
   emptyContainer: {
     marginTop: 116,
```
Comment (nicoback2) [inline packages/mobile/src/screens/search-screen/content/EmptySearch.tsx:15]: nit: use `spacing` here too

Comment (nicoback2) [reply packages/mobile/src/screens/search-screen/content/EmptySearch.tsx:15]: or not, 116 is really big lol

File: packages/mobile/src/store/search/slice.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  storage: AsyncStorage
+}
+
+const persistedSearchReducer = persistReducer(
+  searchPersistConfig,
+  slice.reducer
+)
+export default persistedSearchReducer
+
+// export default slice.reducer
```
Comment (nicoback2) [inline packages/mobile/src/store/search/slice.ts:76]: delete

File: packages/mobile/src/store/store.ts [hunk 3]
```diff
@@ -57,29 +58,35 @@ export type AppState = CommonState & {
...
-    keyboard,
-    oauth,
-    offlineDownloads,
-    remoteConfig,
-    search,
-    mobileUi,
-    walletConnect,
-    shareToStoryProgress
-  })
+// const persistConfig = {
```
Comment (nicoback2) [inline packages/mobile/src/store/store.ts:83]: Did you leave this here intentionally?

**Review by nicoback2 — APPROVED** [review]
awesome work, excited about redux persist :)

File: packages/mobile/package.json [hunk 4]
```diff
@@ -141,6 +141,7 @@
     "react-redux": "8.0.5",
     "react-use": "17.3.2",
     "redux": "4.1.1",
+    "redux-persist": "^6.0.0",
```
Comment (sliptype) [inline packages/mobile/package.json:144]: Pin pls!

File: packages/mobile/src/screens/search-screen/SearchHistoryListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+type SearchHistoryListItemProps = {
+  text: string
+}
+
+export const SearchHistoryListItem = (props: SearchHistoryListItemProps) => {
+  const { text } = props
+  const styles = useStyles()
+  const navigation = useNavigation()
+  const backgroundColor = useColor('neutralLight8')
```
Comment (sliptype) [inline packages/mobile/src/screens/search-screen/SearchHistoryListItem.tsx:41]: I kinda prefer 

```ts
const { neutralLight4, neutralLight8 } = useThemeColors()
```

but up to you!

File: packages/mobile/src/screens/search-screen/SearchScreen.tsx [hunk 6]
```diff
@@ -37,53 +36,46 @@ const useStyles = makeStyles(({ spacing }) => ({
...
-    'playlists',
-    'albums'
-  ])
-  const hasResults = Object.values(searchResults).some(
-    (result) => result && result.length > 0
-  )
+  const searchBarText = useSelector(getSearchBarText)
+  const hasResults = useSelector((state) => {
+    const { tracks, users, playlists, albums } = getSearch(state)
+    return [tracks, users, playlists, albums].some(
```
Comment (sliptype) [inline packages/mobile/src/screens/search-screen/SearchScreen.tsx:42]: Much easier to understand!

File: packages/mobile/src/store/search/slice.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+export const { updateQuery, setHistory, clearHistory, addItem, fetchHistory } =
+  slice.actions
+
+const searchPersistConfig = {
+  key: 'search',
+  storage: AsyncStorage
+}
+
+const persistedSearchReducer = persistReducer(
```
Comment (sliptype) [inline packages/mobile/src/store/search/slice.ts:70]: 🔥

**Review by sliptype — APPROVED** [review]
This is really cool!

File: packages/mobile/src/screens/search-screen/content/EmptySearch.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/search-screen/content/EmptySearch.tsx:15]: yeah with these really random values that existed i kinda throw my hands :D

File: packages/mobile/src/store/store.ts (hunk 3 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/store.ts:83]: no i didnt! i thought initially you had to persist the root all the way down, but looks like can just add it anywhere, so clutch

**Pull request merged**